os/net/ipv6/sicslowpan.c

```diff
@@ -1588,13 +1568,23 @@ output(const linkaddr_t *localdest)
 
 /* Calculate NETSTACK_FRAMER's header length, that will be added in the NETSTACK_MAC */
   packetbuf_set_addr(PACKETBUF_ADDR_RECEIVER, &dest);
-  framer_hdrlen = NETSTACK_FRAMER.length();
-  if(framer_hdrlen < 0) {
-   /* Framing failed, we assume the maximum header length */
-   framer_hdrlen = MAC_MAX_HEADER;
-  }
+#if LLSEC802154_USES_AUX_HEADER
+  /* copy LLSEC level */
+  packetbuf_set_attr(PACKETBUF_ATTR_SECURITY_LEVEL,
+    uipbuf_get_attr(UIPBUF_ATTR_LLSEC_LEVEL));
+#if LLSEC802154_USES_EXPLICIT_KEYS
+  packetbuf_set_attr(PACKETBUF_ATTR_KEY_INDEX,
+    uipbuf_get_attr(UIPBUF_ATTR_LLSEC_KEY_ID));
+#endif /* LLSEC802154_USES_EXPLICIT_KEYS */
+#endif /*  LLSEC802154_USES_AUX_HEADER */
 
-  mac_max_payload = MAC_MAX_PAYLOAD - framer_hdrlen;
+  mac_max_payload = NETSTACK_MAC.max_payload();
```

Comment:
so this results in the exact same value as before, right?

Reply:
That is the intended semantics of the max_payload() function yes!